src/fileio/impls/netcdf/nc_format.cxx

```diff
@@ -352,6 +352,136 @@ bool NcFormat::setRecord(int t) {
   return true;
 }
 
+// Add a variable to the file
+bool NcFormat::addVar_int(const string &name, bool repeat) {
+  if(!is_valid())
+    return false;
+
+  // Create an error object so netCDF doesn't exit
+#ifdef NCDF_VERBOSE
+  NcError err(NcError::verbose_nonfatal);
+#else
+  NcError err(NcError::silent_nonfatal);
+#endif
+
+  NcVar* var;
+  if (!(var = dataFile->get_var(name.c_str()))) {
+    // Variable not in file, so add it.
+    if (repeat)
+      var = dataFile->add_var(name.c_str(), ncInt, 1, recDimList);
+    else
+      var = dataFile->add_var(name.c_str(), ncInt, 0, dimList);
+
+    if(!var->is_valid()) {
+      output_error.write("ERROR: NetCDF could not add int '%s' to file '%s'\n", name.c_str(), fname);
+      return false;
+    }
+  }
+  return true;
+}
+
+bool NcFormat::addVar_BoutReal(const string &name, bool repeat) {
+  if(!is_valid())
+    return false;
+
+  // Create an error object so netCDF doesn't exit
+#ifdef NCDF_VERBOSE
+  NcError err(NcError::verbose_nonfatal);
+#else
+  NcError err(NcError::silent_nonfatal);
+#endif
+
+  NcVar* var;
+  if (!(var = dataFile->get_var(name.c_str()))) {
+    // Variable not in file, so add it.
+    if(lowPrecision) {
+      if (repeat)
+        var = dataFile->add_var(name.c_str(), ncFloat, 1, recDimList);
```

Comment:
You could use a similar pattern for `repeat` as I suggested above for `lowPrecision` to reduce the repeated code here and below

Reply:
Good tip :+1:
FYI, this pattern doesn't work for Ncxx4. In that interface ncFloat and ncDouble have different types, so the ternary doesn't compile.